Userland/Libraries/LibWeb/Fetch/Fetching/Fetching.cpp

```diff
@@ -650,11 +654,33 @@ void fetch_response_handover(JS::Realm& realm, Infrastructure::FetchParams const
     }
     // 7. Otherwise:
     else {
-        // FIXME: 1. Let transformStream be a new TransformStream.
-        // FIXME: 2. Let identityTransformAlgorithm be an algorithm which, given chunk, enqueues chunk in transformStream.
-        // FIXME: 3. Set up transformStream with transformAlgorithm set to identityTransformAlgorithm and flushAlgorithm set
-        //           to processResponseEndOfBody.
-        // FIXME: 4. Set internalResponse’s body’s stream to the result of internalResponse’s body’s stream piped through transformStream.
+        HTML::TemporaryExecutionContext const execution_context { Bindings::host_defined_environment_settings_object(realm), HTML::TemporaryExecutionContext::CallbacksEnabled::Yes };
+
+        // 1. Let transformStream be a new TransformStream.
+        auto transform_stream = MUST(Streams::TransformStream::construct_impl(realm));
+
+        // 2. Let identityTransformAlgorithm be an algorithm which, given chunk, enqueues chunk in transformStream.
+        auto identity_transform_algorithm = JS::create_heap_function(realm.heap(), [&realm, transform_stream](JS::Value chunk) -> WebIDL::ExceptionOr<JS::NonnullGCPtr<WebIDL::Promise>> {
+            TRY(transform_stream->controller()->enqueue(chunk));
+            return WebIDL::create_resolved_promise(realm, JS::js_undefined());
+        });
+
+        // 3. Set up transformStream with transformAlgorithm set to identityTransformAlgorithm and flushAlgorithm set
```

Comment:
I'm not sure this is exactly to spec and if that results in an observable difference. It looks like "Set up" is itself an algorithm: https://streams.spec.whatwg.org/#transformstream-set-up which I can't seem to find. But `construct_impl` does some of this for us, so maybe " be a new TransformStream" is just allocating the object itself?

Reply:
Just allocating it won’t work, it needs to run at least some of the construction steps

Reply:
Is it still required after using "Set up" steps in: https://streams.spec.whatwg.org/#transformstream-set-up

It looks like both `construct_impl` (which is https://streams.spec.whatwg.org/#ts-construct) and "set up" (https://streams.spec.whatwg.org/#transformstream-set-up) both set the size algorithms as an example

Reply:
I’ll have a look at the setup method though, probably overlooked that

Reply:
Thanks for noticing this, Shannon!